### ISSUE TITLE
`doom install` in docker container as root fails with error

### ISSUE THREAD

**Issue by PlaneTraveller** [doomemacs/doomemacs#6754]
### What did you expect to happen?

I'm constructing a containerized doom emacs environment with docker. 

Here's the docker file:
```
# syntax=docker/dockerfile:1

# Installing Archlinux
FROM archlinux
RUN pacman -Sy
RUN pacman -Syyu --noconfirm
RUN pacman -S git ripgrep fd fish neovim emacs xorg --noconfirm

# Installing Doom Emacs
RUN git clone --depth 1 https://github.com/doomemacs/doomemacs ~/.emacs.d
RUN ~/.emacs.d/bin/doom install
ENV PATH=~/.emacs.d/bin:$PATH
```

### What actually happened?

Everything went fine until `doom install` errored out:
```
 => ERROR [7/8] RUN doom install                                                                                                                                                                                                  0.2s
------
 > [7/8] RUN doom install:
#10 0.166 /bin/sh: line 1: doom: command not found
------
executor failed running [/bin/sh -c doom install]: exit code: 127
➜  DockedEmacs docker build .
[+] Building 0.4s (10/11)
 => [internal] load build definition from Dockerfile                                                                                                                                                                              0.0s
 => => transferring dockerfile: 545B                                                                                                                                                                                              0.0s
 => [internal] load .dockerignore                                                                                                                                                                                                 0.0s
 => => transferring context: 2B                                                                                                                                                                                                   0.0s
 => [internal] load metadata for docker.io/library/archlinux:latest                                                                                                                                                               0.0s
 => [1/8] FROM docker.io/library/archlinux                                                                                                                                                                                        0.0s
 => CACHED [2/8] RUN pacman -Sy                                                                                                                                                                                                   0.0s
 => CACHED [3/8] RUN pacman -Syyu --noconfirm                                                                                                                                                                                     0.0s
 => CACHED [4/8] RUN pacman -S git ripgrep fd fish neovim emacs xorg --noconfirm                                                                                                                                                  0.0s
 => CACHED [5/8] RUN echo "PlanarDeck" > /etc/hostname                                                                                                                                                                            0.0s
 => CACHED [6/8] RUN git clone --depth 1 https://github.com/doomemacs/doomemacs ~/.emacs.d                                                                                                                                        0.0s
 => ERROR [7/8] RUN ~/.emacs.d/bin/doom install                                                                                                                                                                                   0.4s
------
 > [7/8] RUN ~/.emacs.d/bin/doom install:
#10 0.350 Debugger entered--Lisp error: (wrong-type-argument number-or-marker-p nil)
#10 0.350   =(0 nil)
#10 0.350   (if (= 0 (file-attribute-user-id (file-attributes user-emacs-directory))) nil (message (concat "Error: this script was executed as root, which is ..." "It will cause file permissions errors later, when ..." "user.\n\n" "If this really *is* what you want, then change the..." "config to root:\n\n" "  chown root:root -R " (abbreviate-file-name user-emacs-directory) "\n\n" "Aborting...")) (kill-emacs 2))
#10 0.350   (progn (if (= 0 (file-attribute-user-id (file-attributes user-emacs-directory))) nil (message (concat "Error: this script was executed as root, which is ..." "It will cause file permissions errors later, when ..." "user.\n\n" "If this really *is* what you want, then change the..." "config to root:\n\n" "  chown root:root -R " (abbreviate-file-name user-emacs-directory) "\n\n" "Aborting...")) (kill-emacs 2)))
#10 0.350   (if (equal (user-real-uid) 0) (progn (if (= 0 (file-attribute-user-id (file-attributes user-emacs-directory))) nil (message (concat "Error: this script was executed as root, which is ..." "It will cause file permissions errors later, when ..." "user.\n\n" "If this really *is* what you want, then change the..." "config to root:\n\n" "  chown root:root -R " (abbreviate-file-name user-emacs-directory) "\n\n" "Aborting...")) (kill-emacs 2))))
#10 0.350   eval-buffer(#<buffer  *load*> nil "/root/.emacs.d/bin/doom" nil t)  ; Reading at buffer position 5578
#10 0.350   load-with-code-conversion("/root/.emacs.d/bin/doom" "/root/.emacs.d/bin/doom" nil t)
#10 0.350   load("/root/.emacs.d/bin/doom" nil t)
#10 0.350   command-line-1(("--no-splash" "--load" "/root/.emacs.d/bin/doom" "--" "install"))
#10 0.350   command-line()
#10 0.350   normal-top-level()
#10 0.350
------
executor failed running [/bin/sh -c ~/.emacs.d/bin/doom install]: exit code: 255
```

It seems that the problem is with installing in `/root`, but this is actually intended because this is a container. It would't make sense to have usual user groups and stuff...

### Describe your attempts to resolve the issue

I've tried running `chown root:root -R ~/.emacs.d` before doom install, but the error is the same. I don't know if this is what the error msg tried to suggest.

If you could think of any easy fix, it would be great! It does't matter if it's a hack or anything.

### Steps to reproduce

Use the dockerfile provided above and run `docker build .` (provided that you're in the same directory as the docker file is).

### System Information

not applicable, doom isn't even installed. The host machine is a mbp 16,4.

**Comment by hlissner** [comment]
Would you mind reporting the output of the following commands, run within the docker container?

```sh
$ emacs --batch --eval '(princ user-emacs-directory)'
```

And

```sh
$ emacs --batch --eval '(file-attributes user-emacs-directory)'
```

**Comment by PlaneTraveller** [comment]
Here's the results:
```
#10 [ 7/10] RUN emacs --batch --eval '(princ user-emacs-directory)'
#10 0.258 ~/.emacs.d/#10 DONE 0.3s

#11 [ 8/10] RUN emacs --batch --eval '(princ (file-attributes user-emacs-directory))'
#11 0.253 (t 10 0 0 (25368 41551 242650 983000) (25368 41553 962650 979000) (25368 41553 962650 979000) 4096 drwxr-xr-x t 535771 144)#11 DONE 0.3s
```

**Comment by hlissner** [comment]
Odd. Both returned the expected results. Could you try these two instead?

```elisp
$ emacs -q --no-site-file --batch --eval '(princ user-emacs-directory)'

$ emacs -q --no-site-file --batch --eval '(princ (file-attributes user-emacs-directory))'
```

**Comment by PlaneTraveller** [comment]
Here's the result:
```
#10 [ 7/10] RUN emacs -q --no-site-file --batch --eval '(princ user-emacs-directory)'
#10 0.250 ~/.emacs.d/#10 DONE 0.3s

#11 [ 8/10] RUN emacs -q --no-site-file --batch --eval '(princ (file-attributes user-emacs-directory))'
#11 0.231 (t 10 0 0 (25368 41555 958650 976000) (25368 41553 962650 979000) (25368 41553 962650 979000) 4096 drwxr-xr-x t 535771 142)#11 DONE 0.2s
```

**Comment by hlissner** [comment]
I narrowed the issue down to a regression introduced in https://github.com/doomemacs/doomemacs/commit/c05e61536ed9faaef5fe6f7f4351ffb3ab15cc87 and fixed it in ceae154. Let me know if the issue resurfaces and I'll reopen it. Thanks for your help and for bringing this to my attention!
